src/transport/transport.hpp

```diff
@@ -38,6 +38,7 @@
 
 #include "configs/autogen/enums.hpp"
 
+#include <openssl/tls1.h> /* needed for TLSClientParams.tls_[min|max]_level values */
```

Comment:
`TLS*` constants aren't used in this file and `transport.hpp` must not have any dependency on openssl.

Reply:
True, anyway as soon as you want to use `TlsParams` you need the constants to fill `tls_version_min` and `tls_version_max`.